packages/@aws-cdk/aws-s3-assets/README.md

```diff
@@ -95,18 +95,19 @@ method `tryBundle()` which should return `true` if local bundling was performed.
 If `false` is returned, docker bundling will be done:
 
 ```ts
+function tryBundle(outputDir: string, options: BundlingOptions) {
+  const canRunLocally = true // replace with actual logic
+  if (canRunLocally) {
+    // perform local bundling here
+    return true;
+  }
+  return false;
+},
+
 new assets.Asset(this, 'BundledAsset', {
   path: '/path/to/asset',
   bundling: {
-    local: {
-      tryBundle(outputDir: string, options: BundlingOptions) {
-        if (canRunLocally) {
-          // perform local bundling here
-          return true;
-        }
-        return false;
-      },
-    },
+    local: {tryBundle},
```

Comment:
Will this compile in Python too?

Reply:
It passes `rosetta:extract --compile` so it should. However, this entire PR is waiting on a fix in `backup` I'm going to make very soon (which is why build is failing).

Reply:
This won't work actually. If Rosetta doesn't flag this, that's a bug in Rosetta.

You need to declare an explicit class that implements the right interface.

Reply:
@kaizen3031593 for context: https://github.com/aws/aws-cdk/issues/17928

Reply:
Rip rosetta. I can investigate what is going on in `rosetta` later. In fact, I looked back at what I did in `lambda-go` which also [included a bundling function.](https://github.com/aws/aws-cdk/pull/17477/files#diff-b1b250e18cba236b1b6e28b7802a670b5d056e618d79e37684cf07f4b644b55dL177-L188) I recall that `rosetta:extract --compile` did not pass there and what I did was end up making that example a `text` snippet and say that the snippet is only available in TypeScript.


Which I guess is bad for the Python folks because I'm basically saying, sorry this doesn't work in Python.

> You need to declare an explicit class that implements the right interface.

That seems to be what #17928 tried and it did not work.